src/Dfc.CourseDirectory.Web/Controllers/LarsSearchController.cs

```diff
@@ -1,120 +1,111 @@
+                Facets = new[] { nameof(Lars.AwardOrgCode), nameof(Lars.NotionalNVQLevelv2) },
+                PageSize = 0
+            });
 
-                var criteria = _larsSearchHelper.GetLarsSearchCriteria(
-                    requestModel,
-                    _paginationHelper.GetCurrentPageNo(Request.GetDisplayUrl(), _larsSearchSettings.PageParamName),
-                    _larsSearchSettings.ItemsPerPage,
-                    (LarsSearchFacet[])Enum.GetValues(typeof(LarsSearchFacet)));
+            var result = await _searchClient.Search(new LarsSearchQuery
+            {
+                SearchText = request.SearchTerm,
+                NotionalNVQLevelv2Filters = request.NotionalNVQLevelv2Filter,
+                AwardOrgCodeFilters = request.AwardOrgCodeFilter,
+                AwardOrgAimRefFilters = request.AwardOrgAimRefFilter,
+                CertificationEndDateFilter = DateTimeOffset.UtcNow,
+                Facets = new[] { nameof(Lars.AwardOrgCode), nameof(Lars.NotionalNVQLevelv2) },
+                PageSize = _larsSearchSettings.ItemsPerPage,
+                PageNumber = request.PageNo
+            });
 
-                var result = await _larsSearchService.SearchAsync(criteria);
-                if (resultAll.IsSuccess && resultAll.HasValue && result.IsSuccess && result.HasValue)
+            var viewModel = new LarsSearchResultModel
+            {
+                SearchTerm = request.SearchTerm,
+                Items = result.Results.Select(LarsSearchResultItemModel.FromLars),
+                Filters = new[]
                 {
-                    requestModelAll.NotionalNVQLevelv2Filter = requestModel.NotionalNVQLevelv2Filter;
-                    requestModelAll.AwardOrgCodeFilter = requestModel.AwardOrgCodeFilter;
-
-                    var filters = _larsSearchHelper.GetLarsSearchFilterModels(resultAll.Value.SearchFacets, requestModelAll);
-                    var items = _larsSearchHelper.GetLarsSearchResultItemModels(result.Value.Value);
-
-                    var sfSearch = result.Value.SearchFacets;
-                    LarsSearchFilterModel filter = filters.FirstOrDefault(f => f.Title == "Awarding organisation");
-                    foreach (SearchFacet sf in resultAll?.Value?.SearchFacets?.AwardOrgCode)
-                    {
-                        LarsSearchFilterItemModel fim = filter?.Items
-                                                              ?.FirstOrDefault(m => m.Value == sf.Value);
-                        if (fim != null) //&& filter != null)
-                            fim.Count = sfSearch.AwardOrgCode
-                                                .FirstOrDefault(f => f.Value == sf.Value)
-                                               ?.Count ?? 0;
-                    }
+                    new LarsSearchFilterModel("Qualification level", unfilteredResult.Facets[nameof(Lars.NotionalNVQLevelv2)].Select((f, i) =>
+                        new LarsSearchFilterItemModel(
```

Comment:
Explicitly naming parameters here (especially the `string` ones) would be easier to parse, for me.

Reply:
Agreed. Not sure why I didn't do that here.